x/gamm/keeper/migrate.go

```diff
@@ -118,28 +121,51 @@ func (k Keeper) validateRecords(ctx sdk.Context, records []types.BalancerToConce
 		}
 
 		// Ensure the provided balancerPoolId exists and that it is of type balancer
-		pool, err := k.GetPool(ctx, record.BalancerPoolId)
+		balancerPool, err := k.GetPool(ctx, record.BalancerPoolId)
 		if err != nil {
 			return err
 		}
-		poolType := pool.GetType()
+		poolType := balancerPool.GetType()
 		if poolType.String() != "Balancer" {
 			return fmt.Errorf("Balancer pool ID #%d is not of type balancer", record.BalancerPoolId)
 		}
 
 		// Ensure the provided ClPoolId exists and that it is of type concentrated.
 		// If clPoolID is 0, this signals a removal, so we skip this check.
+		var clPool poolmanagertypes.PoolI
 		if record.ClPoolId != 0 {
-			pool, err = k.clKeeper.GetPool(ctx, record.ClPoolId)
+			clPool, err = k.clKeeper.GetPool(ctx, record.ClPoolId)
 			if err != nil {
 				return err
 			}
-			poolType = pool.GetType()
+			poolType = clPool.GetType()
 			if poolType.String() != "Concentrated" {
 				return fmt.Errorf("Concentrated pool ID #%d is not of type concentrated", record.ClPoolId)
 			}
 		}
 
+		// Ensure the balancer pools denoms are the same as the concentrated pool denoms
+		// If clPoolID is 0, this signals a removal, so we skip this check.
```

Comment:
Err, i'm not clear what clPool Id == 0 actually is doing here.

Perhaps lets have a block like
```
if clPoolId == 0 { // do stuff & exit }

// current logic
```

Reply:
We don't want validation to run when a CL pool is equal to 0. This is because when we submit a gov prop with a CL pool equal to zero, it indicates we want to remove the link between the CL and balancer pool.

I feel like what we do currently in this PR is equal to what you are saying. If clPoolId == 0, do stuff and exit (in this case, the stuff to do is nothing and we exit). I might be misunderstanding though.

Reply:
I propose to merge this PR and resolve this discussion separately unless any concerns.

I think changing this logic would require effort outside of the scope of work done here. So it is better to be addressed in a subsequent PR